### PR TITLE
feat(select) Keyboard navigation

### LINKED ISSUE #185: Select: Manage keyboard navigation
To navigate inside the widget

### PR DESCRIPTION
**Pull request by fbasso**
fix #185

### CODE REVIEW

**Comment by quentinderoubaix** [comment]
As discussed, please update the classes for the badge hover and add an accessibility test when opening the select popup.

File: core/src/components/select/select.ts [hunk 1]
```diff
@@ -547,6 +555,18 @@ export function createSelect<Item>(config?: PropsConfig<SelectProps<Item>>): Sel
...
 				});
 			},
+
+			onRemoveBadgeClick(event: any, item: Item) {
+				const element = event.target;
+				refreshElements();
+				widget.api.unselect(item);
+				// Waiting for refresh by the framework, to have the elements inside or outside the dom
+				setTimeout(() => {
+					focusLeft({event, referenceElement: element}) || focusRight({event, referenceElement: element});
```
[the remark this answers is not in the document]
Comment (fbasso) [reply core/src/components/select/select.ts:565]: Well, focusLeft comes from the `createEventManager`. I'm not supposed to know what is done with the event, no ? Even if it's used only to perform a preventDefault, it's probably safer if, for example, implementation changes in the future.

What do you think ?

File: react/demo/src/app/samples/select/Custom.route.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const SlotBadgeLabel = ({itemContext, widget}: SelectItemContext<WikiResult>) =>
 				type="button"
 				className="btn-close ms-1 wiki-btn-close"
 				aria-label="Close"
-				onClick={() => widget.api.unselect(itemContext.item)}
+				onClick={(e) => widget.actions.onRemoveBadgeClick(e as any, itemContext.item)}
```
[the remark this answers is not in the document]
Comment (fbasso) [reply react/demo/src/app/samples/select/Custom.route.tsx:36]: To be discussed what to do... 

Typescript is quite annoying on this point, that's why I used as any in both side (declaration + usage). If I remove `as any`: Argument of type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to parameter of type 'MouseEvent'

And if I give this exact type in the function type, I have: Type 'MouseEvent' is not generic

Do you know what to do ?

File: core/src/components/select/select.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fbasso) [reply core/src/components/select/select.ts:565]: Thnaks for your proposal @quentinderoubaix, I have integrated them ;)

File: react/demo/src/app/samples/select/Custom.route.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (fbasso) [reply react/demo/src/app/samples/select/Custom.route.tsx:36]: Done, thanks !

**Comment by fbasso** [comment]
Thanks for yours comments, @quentinderoubaix. Accessibility concerns will be done in a separate PR.

**Review by quentinderoubaix — APPROVED** [review]
thx for the PR @fbasso ! LGTM

**Pull request merged**